overlord/servicestate/servicemgr.go

```diff
@@ -117,6 +117,34 @@ func (m *ServiceManager) ensureSnapServicesUpdated() (err error) {
 		return nil
 	}
 
+	var sysd systemd.Systemd
```

Comment:
no strong opinion, but conceptually this belongs to snapstate (as it is the manager that creates the units to start with) and not here

Reply:
Do you mean to move the content of the loop after into a method in snapstate, but still call it from ServiceManager.ensureSnapServicesUpdated? Otherwise I am a bit confused.

Reply:
Or did you mean it should be called from `SnapManager.Ensure`?

Reply:
the latter, this part going under SnapManager.Ensure

Reply:
OK. I have managed to get it to work with the spread test. But now I have to modify all the tests in `snapstate_test.go`.

Reply:
why all the tests? maybe I'm misunderstanding, what we do sometimes in these cases is expose a way in export_test.go to set a flag on the manager to avoid some of the Ensure logic when it's not the test focus. 

see SetBootOkRan for example in devicestate/export_test.go

Reply:
any news on this front?

Reply:
Sorry, I forgot to finish it. Now it is pushed.